src/components/Chat/AudioInput/index.js

```diff
@@ -0,0 +1,131 @@
+import React, { memo, useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import RecordRTC from 'recordrtc'
+import Timer from 'react-compound-timer'
+import ButtonContainer from '../../ButtonContainer'
+import './styles.scss'
+
+const AudioInput = props => {
+  const { handleSubmit, closeModal, isRecording } = props
+  const [stream, setStream] = useState(null)
+  const [recorder, setRecorder] = useState(null)
+  const [recordingFinished, setRecorderFinished] = useState(false)
+  const timer = React.createRef()
+
+  // If timer has reached 60 seconds, it sets recordingFinished to true and this is activated
```

Comment:
If you want to combine some useEffects, you cold do for example like this:
```
useEffect(() => {
    if (recordingFinished) {
      timer.current.stop()
      recorder.stopRecording()
      stream.stop()
    } else if (recorder) {
      recorder.startRecording()
    }
  }, [recordingFinished, recorder, stream, timer])
```

And remove this:
```
useEffect(() => {
    if (recorder) {
      recorder.startRecording()
    }
  }, [recorder])
```

Reply:
IMO 3 useEffects is totally fine, and this might be even more clear divided into separate useEffects. I think this is more about personal preference than anything else. But this seems to work nicely! 😎